src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -498,20 +492,17 @@ Response put(URI uri, Object object, boolean newEntity, int writeQuorum) {
      * @param loginUsername
      * @param password
      * @param connectOptions
-     * @param authCookie
      */
     private void doInit(String scheme, String hostname, int port,
-                        String loginUsername, String password, ConnectOptions connectOptions,
-                        String authCookie) {
+                        String loginUsername, String password, ConnectOptions connectOptions) {
 
-        CouchDbProperties props;
-        if (authCookie == null) {
-            props = new CouchDbProperties(scheme, hostname, port,
-                    loginUsername, password);
+        CouchDbProperties props = new CouchDbProperties(scheme, hostname, port);
+        if (loginUsername != null && password != null) {

```

Comment:
Yea, we could use `CouchDbException` and throw a message like:
"Invalid username and/or password.  Please check the credentials and try again."